homeassistant/components/light/limitlessled.py

```diff
@@ -159,10 +160,22 @@ def turn_on(self, **kwargs):
         if ATTR_XY_COLOR in kwargs:
             self._xy_color = kwargs[ATTR_XY_COLOR]
 
-        self.pool.execute(self.controller_id, "set_color",
-                          self._xy_to_led_color(self._xy_color), self.group)
+        effect = kwargs.get(ATTR_EFFECT)
+
+        if effect:
```

Comment:
We could consolidate these `if` statements. Effect will be `None` if not given so no need to check if exists.
```python
if effect == EFFECT_COLORLOOP:
  …
elif …:
  …
else:
  …
```